src/segment_evaluation/mod.rs

```diff
@@ -115,45 +116,45 @@ fn belong_to_segment(
     operator: &str,
     reference_value: &str,
 ) -> std::result::Result<bool, CheckOperatorErrorDetail> {
     match operator {
         "is" => match attribute_value {
-            AttrValue::String(data) => Ok(*data == reference_value),
-            AttrValue::Boolean(data) => Ok(*data == reference_value.parse::<bool>()?),
-            AttrValue::Numeric(data) => Ok(*data == reference_value.parse::<f64>()?),
+            Value::String(data) => Ok(*data == reference_value),
+            Value::Boolean(data) => Ok(*data == reference_value.parse::<bool>()?),
+            Value::Numeric(data) => Ok(*data == reference_value.parse()?),
         },
         "contains" => match attribute_value {
-            AttrValue::String(data) => Ok(data.contains(reference_value)),
+            Value::String(data) => Ok(data.contains(reference_value)),
             _ => Err(CheckOperatorErrorDetail::StringExpected),
         },
         "startsWith" => match attribute_value {
-            AttrValue::String(data) => Ok(data.starts_with(reference_value)),
+            Value::String(data) => Ok(data.starts_with(reference_value)),
             _ => Err(CheckOperatorErrorDetail::StringExpected),
         },
         "endsWith" => match attribute_value {
-            AttrValue::String(data) => Ok(data.ends_with(reference_value)),
+            Value::String(data) => Ok(data.ends_with(reference_value)),
             _ => Err(CheckOperatorErrorDetail::StringExpected),
         },
         "greaterThan" => match attribute_value {
             // TODO: Go implementation also compares strings (by parsing them as floats). Do we need this?
             //       https://github.com/IBM/appconfiguration-go-sdk/blob/master/lib/internal/models/Rule.go#L82
             // TODO: we could have numbers not representable as f64, maybe we should try to parse it to i64 and u64 too?
-            AttrValue::Numeric(data) => Ok(*data > reference_value.parse()?),
+            Value::Numeric(data) => Ok(*data > reference_value.parse()?),
             _ => Err(CheckOperatorErrorDetail::EntityAttrNotANumber),
         },
         "lesserThan" => match attribute_value {
-            AttrValue::Numeric(data) => Ok(*data < reference_value.parse()?),
+            Value::Numeric(data) => Ok(*data < reference_value.parse()?),
             _ => Err(CheckOperatorErrorDetail::EntityAttrNotANumber),
         },
         "greaterThanEquals" => match attribute_value {
-            AttrValue::Numeric(data) => Ok(*data >= reference_value.parse()?),
+            Value::Numeric(data) => Ok(*data >= reference_value.parse()?),
             _ => Err(CheckOperatorErrorDetail::EntityAttrNotANumber),
         },
         "lesserThanEquals" => match attribute_value {
-            AttrValue::Numeric(data) => Ok(*data <= reference_value.parse()?),
+            Value::Numeric(data) => Ok(*data <= reference_value.parse()?),
             _ => Err(CheckOperatorErrorDetail::EntityAttrNotANumber),
         },
```

Comment:
This behaves differently now:
previously `data` was a `f64` and `reference_value` a string being parsed into a `f64`. Now we have `NumericValue` and the string being parsed also into a `NumericValue`. things like `u64::MAX` will now be compared differently than before.

Reply:
We don't have cross-language tests yet, so this should be fine. It's our "best" effort 😸